test/integration/testExecute.js

```diff
@@ -163,180 +165,249 @@ describe('Execute test', function ()
+
+    const rowAsserts = (testCase, row) => {
+      // Check the column, header, and value is correct
+      const attributeWithPrefix = testCase.attributeNamePrefix + TEST_ATTRIBUTE_NAME;
+
+      if (testCase.name === 'xml_single_element') {
+        assert.strictEqual(row[TEST_COL][TEST_HEADER], TEST_XML_VAL);
+      } else if (testCase.name === 'xml_single_element_with_attribute') {
+        assert.strictEqual(row[TEST_COL][TEST_HEADER][ELEMENT_VALUE_FIELD], TEST_XML_VAL);
+      } else if (testCase.name === 'xml_with_attribute_and_custom_parser') {
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][ELEMENT_VALUE_FIELD], TEST_XML_VAL);
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][attributeWithPrefix], TEST_ATTRIBUTE_VALUE);
+      } else if (testCase.name === 'xml_skip_attributes') {
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER], TEST_XML_VAL);
+        assert.equal(row[TEST_COL]['node'][TEST_HEADER][testCase.attributeNamePrefix + TEST_ATTRIBUTE_NAME], undefined);
+      } else if (testCase.name === 'xml_attribute_only') {
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][ELEMENT_VALUE_FIELD], undefined);
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][testCase.attributeNamePrefix + TEST_ATTRIBUTE_NAME], TEST_ATTRIBUTE_VALUE);
+      } else if (testCase.name === 'xml_attribute_empty_value') {
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][ELEMENT_VALUE_FIELD], undefined);
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][testCase.attributeNamePrefix + TEST_ATTRIBUTE_NAME], TEST_ATTRIBUTE_VALUE);
+      }
+    };
+
+    for (let index = 0; index < testCases.length; index++) {
+      const testCase = testCases[index];
+      it(testCase.name, createItCallback(testCase, rowAsserts));
+    }
+  });
+
+  describe('Variant JSON', function () {
+    const TEST_JSON_VAL = '<123>';
+
+    before(async () => {
+      const createFileFormatVariant = `CREATE OR REPLACE FILE FORMAT ${TEST_VARIANT_FORMAT} TYPE = JSON`;
+      await testUtil.executeCmdAsync(connection, createFileFormatVariant);
+    });
+    //
+    beforeEach(async () => {
+      // Drop stage
+      await testUtil.executeCmdAsync(connection, createStageVariant);
+
+    });
+    afterEach(async () => {
+      // Drop stage
+      await testUtil.executeCmdAsync(connection, truncateTableVariant);
+      await testUtil.executeCmdAsync(connection, dropStageVariant);
+    });
+
+    const testCases =
+      [
+        {
+          name: 'raw_json',
+          type: 'JSON',
+          fileExtension: '.json',
+          sampleData: `{${TEST_HEADER}: \"${TEST_JSON_VAL}\"}`
+        }
+      ];
+
+    const rowAsserts = (testCase, row) => {
+      // Check the column, header, and value is correct
+      if (testCase.name === 'raw_json') {
+        assert.strictEqual(row[TEST_COL][TEST_HEADER], TEST_JSON_VAL);
+      }
+    };
+
+    for (let index = 0; index < testCases.length; index++) {
```

Comment:
maybe let's rewrite it to `testCases.forEach`?

Reply:
ok